services/wallet/router/filter.go

```diff
@@ -0,0 +1,103 @@
+package router
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+func filterRoutesV2(routes [][]*PathV2, amountIn *big.Int, fromLockedAmount map[uint64]*hexutil.Big) [][]*PathV2 {
+	if len(fromLockedAmount) == 0 {
+		return routes
+	}
+
+	routesAfterNetworkCompliance := filterNetworkComplianceV2(routes, fromLockedAmount)
+	return filterCapacityValidationV2(routesAfterNetworkCompliance, amountIn, fromLockedAmount)
+}
+
+// filterNetworkComplianceV2 performs the first level of filtering based on network inclusion/exclusion criteria.
+func filterNetworkComplianceV2(routes [][]*PathV2, fromLockedAmount map[uint64]*hexutil.Big) [][]*PathV2 {
+	var filteredRoutes [][]*PathV2
+	fromIncluded, fromExcluded := setupRouteValidationMapsV2(fromLockedAmount)
+
+	for _, route := range routes {
+		if isValidForNetworkComplianceV2(route, fromIncluded, fromExcluded) {
+			filteredRoutes = append(filteredRoutes, route)
+		}
+	}
+	return filteredRoutes
+}
+
+// isValidForNetworkComplianceV2 checks if a route complies with network inclusion/exclusion criteria.
+func isValidForNetworkComplianceV2(route []*PathV2, fromIncluded, fromExcluded map[uint64]bool) bool {
+	for _, path := range route {
+		if fromExcluded[path.From.ChainID] {
+			return false
+		}
+		if _, ok := fromIncluded[path.From.ChainID]; ok {
+			fromIncluded[path.From.ChainID] = true
+		}
+	}
+
+	for _, included := range fromIncluded {
+		if !included {
+			return false
+		}
+	}
+
+	return true
+}
+
+// setupRouteValidationMapsV2 initializes maps for network inclusion and exclusion based on locked amounts.
+func setupRouteValidationMapsV2(fromLockedAmount map[uint64]*hexutil.Big) (map[uint64]bool, map[uint64]bool) {
```

Comment:
Maybe we should check the validity of `fromLockedAmount` at the moment when `SuggestedRoutesV2` is called. 

We should check that are all entries are positive, I guess that if the user deliberately doesn't want to send any amount from a certain chain there are two options to do that from the UI side:
- disable that chain (which means we should add that chain to `DisabledFromChainIDs`array)
- or set 0 value for locked amount on that chain

Reply:
That's a good idea. We need some better validation generally.

Reply:
@saledjenic I've created an issue for this https://github.com/status-im/status-go/issues/5227